service/src/main/kotlin/fi/espoo/evaka/application/ApplicationQueries.kt

```diff
@@ -115,6 +115,35 @@ fun duplicateApplicationExists(
         .isNotEmpty()
 }
 
+fun activePlacementExists(
+    h: Handle,
+    childId: UUID,
+    type: ApplicationType
+): Boolean {
+    // language=sql
+    val placementTypes = when (type) {
+        ApplicationType.DAYCARE -> listOf(PlacementType.DAYCARE, PlacementType.DAYCARE_PART_TIME)
+        ApplicationType.PRESCHOOL -> listOf(PlacementType.PREPARATORY, PlacementType.PREPARATORY_DAYCARE, PlacementType.PRESCHOOL, PlacementType.PRESCHOOL_DAYCARE)
+        else -> listOf()
+    }
+    if (placementTypes.isEmpty()) return false
+    val sql =
+        """
+            SELECT 1
+            FROM placement
+            WHERE 
+                child_id = :childId AND
+                type IN (<types>) AND 
+                current_date <= end_date
+        """.trimIndent()
+    return h.createQuery(sql)
+        .bind("childId", childId)
+        .bindList("types", placementTypes)
```

Comment:
The `bindList` method has some quirks so regular `bind` with an array is preferable, you'll have to change the condition to `type = ANY(:types)` as well.